docs: better explain how environment variables work in production (#27189)
nuxt/nuxt: docs/2.guide/2.directory-structure/2.env.md

When updating `.env` in development mode, the Nuxt instance is automatically restarted to apply new values to the `process.env`.

## Production

**After your server is built**, you are responsible for setting environment variables when you run the server.

Your `.env` files will not be read at this point. How you do this is different for every environment.

This design decision was made to ensure compatibility across various deployment environments, some of which may not have a traditional file system available, such as serverless platforms or edge networks like Cloudflare Workers.

Since `.env` files are not used in production, you must explicitly set environment variables using the tools and methods provided by your hosting environment. Here are some common approaches:

> * Pass the environment variables as arguments using the terminal

Reviewer: ```suggestion
* You can pass the environment variables as arguments using the terminal:
```

   `$ DATABASE_HOST=mydatabaseconnectionstring node .output/server/index.mjs`

> * Shell Configuration Files
> 
>    You can set environment variables in shell configuration files like `.bashrc` or `.profile`.

Reviewer: ```suggestion
* You can set environment variables in shell configuration files like `.bashrc` or `.profile`.
```

> * Cloud Service Interfaces
> 
>    Many cloud service providers like Vercel, Netlify, and AWS provide interfaces for setting environment variables via their dashboards, CLI tools or configuration files.

Reviewer: ```suggestion
* Many cloud service providers, such as Vercel, Netlify, and AWS, provide interfaces for setting environment variables via their dashboards, CLI tools or configuration files.
```

When updating `.env` in development mode, the Nuxt instance is automatically restarted to apply new values to the `process.env`.

## Production

**After your server is built**, you are responsible for setting environment variables when you run the server.

Your `.env` files will not be read at this point. How you do this is different for every environment.

This design decision was made to ensure compatibility across various deployment environments, some of which may not have a traditional file system available, such as serverless platforms or edge networks like Cloudflare Workers.

Since `.env` files are not used in production, you must explicitly set environment variables using the tools and methods provided by your hosting environment. Here are some common approaches:

> ### Pass the environment variables as arguments using the terminal

Reviewer 2: Bulleted list might be a better fit for these three headings, what do you think?

Author: I agree, pushed changes.

```bash [Terminal]
DATABASE_HOST=mydatabaseconnectionstring node .output/server/index.mjs

> ```

Reviewer: This (and the other sections can probably be indented three spaces so they are nested under the list items

Author: Indented the sections, pushed changes.
